src/Ng/CommandHelpers.cs

```diff
@@ -410,64 +200,58 @@ public static Lucene.Net.Store.Directory GetCopyDestLuceneDirectory(IDictionary<
         {
             IDictionary<string, string> names = new Dictionary<string, string>
             {
-                { Constants.DirectoryType, Constants.DestDirectoryType },
-                { Constants.Path, Constants.DestPath },
-                { Constants.StorageAccountName, Constants.DestStorageAccountName },
-                { Constants.StorageKeyValue, Constants.DestStorageKeyValue },
-                { Constants.StorageContainer, Constants.DestStorageContainer }
+                { Arguments.DirectoryType, Arguments.DestDirectoryType },
+                { Arguments.Path, Arguments.DestPath },
+                { Arguments.StorageAccountName, Arguments.DestStorageAccountName },
+                { Arguments.StorageKeyValue, Arguments.DestStorageKeyValue },
+                { Arguments.StorageContainer, Arguments.DestStorageContainer }
             };
 
             return GetLuceneDirectoryImpl(arguments, names, required);
         }
 
-        public static Lucene.Net.Store.Directory GetLuceneDirectoryImpl(IDictionary<string, string> arguments, IDictionary<string, string> names, bool required = true)
+        public static Lucene.Net.Store.Directory GetLuceneDirectoryImpl(IDictionary<string, string> arguments, IDictionary<string, string> argumentNameMap, bool required = true)
         {
             try
             {
-                string luceneDirectoryType;
-                TryGetArgument(arguments, names[Constants.DirectoryType], out luceneDirectoryType, required: true);
+                var luceneDirectoryType = arguments.GetOrNull(argumentNameMap[Arguments.DirectoryType]);
```

Comment:
this actually still does (the if statement below, with `luceneDirectoryType.Equals`, will crash if the directory type is null/undefined) but i will browse over the file and verify that every `required` is handled properly.